viewing_party/party.py

```diff
@@ -0,0 +1,204 @@
+
+    # find the difference between sets
+    set_unique_user_watched = set_user_watched - set_friends_watched
+
+    # add unique movie into result_list 
+    for unique_movie in set_unique_user_watched:
+        for movie in user_data["watched"]:
+            if movie["title"] == unique_movie:
+                list_unique_movie_user_watched.append(movie)
+    return list_unique_movie_user_watched
+
+
+    # Alternative Approach 
+
+    # list_unique_movie_user_watched = []
+    
+    # set_user_watched = set()
+    # for movie_user_watched in user_data["watched"]:
+    #     movie_userdict_value_hashable = frozendict(movie_user_watched)
+    #     if movie_userdict_value_hashable not in set_user_watched:                 # put dict_value in set How to add mutable elements to set?
+    #         set_user_watched.add(movie_userdict_value_hashable)
+
+    # set_friends_watched = set()
+    # for friends_data in user_data["friends"]:
+    #     for movie in friends_data["watched"]:
+    #         movie_friendsdict_value_hashable = frozendict(movie) 
+    #         set_friends_watched.add(movie_friendsdict_value_hashable)
+
+    # # find the difference between sets
+    # set_unique_user_watched = set_user_watched - set_friends_watched
+    # return list(set_unique_user_watched)
+
+
+def get_friends_unique_watched(user_data):
+    set_user_watched = set()
+    set_friends_watched = set()
+    list_unique_friends_watched = []
+
+    for friends_data in user_data["friends"]:
+        helper_func_add_movie_title_to_set(set_friends_watched,friends_data)
+
+    helper_func_add_movie_title_to_set(set_user_watched,user_data)
+    
+    set_unique_friends_watched = set_friends_watched - set_user_watched
+
+    #append dict into list 
+    for unique_movie in set_unique_friends_watched:
+        for friends_data in user_data["friends"]:
+            for movie in friends_data["watched"]:
+                if movie["title"] == unique_movie and movie not in list_unique_friends_watched:         # & ? and??
+                    list_unique_friends_watched.append(movie)
+    return list_unique_friends_watched
+
+
+
+# Wave 4
+def get_available_recs(user_data):
+    list_recommended_movies = []
+    list_unique_friends_watched = get_friends_unique_watched(user_data)
+
+    for unique_movie in list_unique_friends_watched:
+        for service_provider in user_data["subscriptions"]:
+            if unique_movie["host"] == service_provider:
+                list_recommended_movies.append(unique_movie)
+    return list_recommended_movies
+
+
+# Wave 5
+def get_new_rec_by_genre(user_data):
+    list_recommended_movies = []
+    list_unique_friends_watched = get_friends_unique_watched(user_data)
+    most_watched_genre = get_most_watched_genre(user_data)
+
+    for unique_movie in list_unique_friends_watched:
+        if unique_movie["genre"] == most_watched_genre:   # helper function possible to combine with below func?
```

Comment:
Just to answer your comment, I'm sure you can where either you are returning the recommendations for movies by genre or returning the recommendations for movies by favorites